internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -486,6 +512,47 @@ func (b *BeatsMonitor) monitoringNamespace() string {
 	return defaultMonitoringNamespace
 }
 
+func (b *BeatsMonitor) getCollectorTelemetryEndpoint() string {
+	type metricsReaderConfig struct {
+		Pull struct {
+			Exporter struct {
+				Prometheus struct {
+					Host string `mapstructure:"host"`
+					Port int    `mapstructure:"port"`
+				} `mapstructure:"prometheus"`
+			} `mapstructure:"exporter"`
+		} `mapstructure:"pull"`
+	}
+	var otel struct {
+		Service struct {
+			Telemetry struct {
+				Metrics struct {
+					Readers []metricsReaderConfig `mapstructure:"readers"`
+				} `mapstructure:"metrics"`
+			} `mapstructure:"telemetry"`
+		} `mapstructure:"service"`
+	}
+	if err := b.otelConfig.Unmarshal(&otel, confmap.WithIgnoreUnused()); err == nil {
+		for _, reader := range otel.Service.Telemetry.Metrics.Readers {
+			p := reader.Pull.Exporter.Prometheus
+			if p.Host != "" || p.Port != 0 {
+				host := "localhost"
+				port := 8888
```

Comment:
Is there any way to get this to use a named pipe/unix socket instead?

If we can't do that, the port will need to be configurable and to keep the `--develop` option working to allow two agents on the same host is has to be a random free port. https://github.com/pkoutsovasilis/elastic-agent/blob/c99794996e768f038b485939b1cd02aafd5db724/internal/pkg/agent/configuration/grpc.go#L41-L46

If we could just default this to a random free port initially it would be easiest, with the option to specify a fixed for cases where users need to work around local firewall rules.

Reply:
Sockets would be nicer but it doesn't seem possible with the current Collector code? Unless I misunderstood the outcome of the beats receiver meetings, that example might use grpc but collector configs don't yet support non-http urls (and in any case that wouldn't work with metricbeat -- though hopefully we'll be dumping the metricbeat approach for 9.3 anyway).

I agree 8888 will run into conflicts, but that's already a problem independent of this PR (which is why this PR checks for alternate port configs).

Reply:
I've been down this rabbit hole a few times - https://github.com/prometheus/prometheus/issues/12024 implies it's open for discussion but not implemented.  https://github.com/prometheus/node_exporter/issues/759 for example uses this as a reason not to implement support

Reply:
Using a fixed port is basically unreleasable IMO and would be a breaking change for anyone that already has something on this port.

We can use whatever transport we have to but we have to get away from specifying the fixed port here and requiring user intervention to change it. We would have to build a UI component to change the port in Fleet for example for that to even be viable.

Reply:
> I've been down this rabbit hole a few times - prometheus/prometheus#12024 implies it's open for discussion but not implemented.

This almost looks like it's just waiting for someone to say they are willing implement it.